### DISCUSSION TITLE
How to show prompt on route change in Next 13?

### DISCUSSION THREAD

**Discussion by nucleartux** [vercel/next.js#48564] [Help]
### Summary

I want to show confirmation modal if I have some unsaved information on a page.
How I can do it using Next 13?
Similar to this react-router library https://github.com/sshyam-gupta/react-router-prompt

**Comment by haydenbleasel** [comment]
Hi @nucleartux,

As router events are not in next/navigation yet, I haven't tested this but you might try writing a provider component in your root `layout.tsx` e.g.

```
<RouteChangeProvider>
  {children}
</RouteChangeProvider>
```

then using that to set up a useEffect that utilises changes in the next/navigation pathname e.g.

```
import { usePathname } from 'next/navigation';
import { useEffect } from 'react';

const RouteChangeProvider = () => {
  const pathname = usePathname();

  useEffect(() => {
    // do stuff on pathname change
  }, [pathname]);
}

export default RouteChangeProvider;
```

**Reply by nucleartux** [reply]
Thank you, but I suppose this code simply listens for path changes without the ability to prevent route changes.

**Reply by haydenbleasel** [reply]
Sorry @nucleartux I posted a potential solution to the first part of the issue while I was simultaneously looking for a way of blocking history changes at the window-level within that hook. I'm not sure there's anything exported from `next` that will help with that unfortunately 🤔
